File: runtimes/org.eclipse.edt.runtime.java.jee/src/eglx/lang/Resources.java
/*******************************************************************************
 * Copyright © 2012 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 * IBM Corporation - initial API and implementation
 *
 *******************************************************************************/
package eglx.lang;

import java.net.URI;
import java.net.URISyntaxException;

import org.eclipse.edt.javart.messages.Message;
import org.eclipse.edt.javart.resources.egldd.RuntimeDeploymentDesc;

import resources.edt.binding.BindingResourceProcessor;

public class Resources {
	private static BindingResourceProcessor bindingProcessor;

	/**
	 * get the resource binding from the egldd
	 */
	public static Object getResource(String uriStr)  throws AnyException{
		try {
			URI uri = new URI(uriStr);
			if("resource".equals(uri.getScheme())){
				String query = uri.getQuery() != null && uri.getQuery().length() > 0 ? "?" + uri.getQuery() :"";
				String fragment = uri.getFragment() != null && uri.getFragment().length() > 0 ? "#" + uri.getFragment() :"";
				uri = new URI(uri.getSchemeSpecificPart() + query + fragment);
			}
			if("binding".equals(uri.getScheme())){
				return getBindingResourceProcessor().resolve(uri);
			}
		} catch (URISyntaxException e) {
		}
		AnyException ae = new AnyException();
		throw ae.fillInMessage( Message.RESOURCE_NO_PROCESSOR, uriStr );
	}
	
	private static BindingResourceProcessor getBindingResourceProcessor(){
		if(bindingProcessor == null){
			bindingProcessor = new BindingResourceProcessor();
		}
		return bindingProcessor;
	}
	public static void setBindingResourceProcessor(BindingResourceProcessor ideBindingProcessor) {
		bindingProcessor = ideBindingProcessor;
	}
	
	public static interface ResourceLocator {
		public RuntimeDeploymentDesc getDeploymentDesc(URI propertyFileURI);
	}

}

File: runtimes/org.eclipse.edt.runtime.javascript/runtime/eglx/lang/Resources.js
/*******************************************************************************
 * Copyright © 2012 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 * IBM Corporation - initial API and implementation
 *
 *******************************************************************************/
egl.defineClass(
	'eglx.lang', 'Resources',
{
	"constructor" : function() {		
	}
});

egl.eglx.lang.Resources["getResource"] = function(uri) {
	var ret = undefined;
	var idx = uri.indexOf("resource:");
	if(idx === 0){
		uri = uri.slice(idx + 9);//pick up everything paste resource:
	}
	idx = uri.indexOf("binding:");
	if(idx === -1){
		throw egl.eglx.services.createServiceInvocationException("CRRUI3663E", [uri]);  
	}
	else{
		uri = uri.slice(idx + 8);//pick up everything paste binding:
	}
	var bindingKey;
	idx = uri.indexOf("file:");
	if(idx === -1){
		eglddName = egl__defaultDeploymentDescriptor;
		bindingKey = uri;//uri is the binding key
	}
	else{
		uri = uri.slice(idx + 5);//pick up everything paste file:
		idx = uri.indexOf("#");
		eglddName = uri.slice(0, idx);//pick up everything up to the first # as the file name:
		bindingKey = uri.slice(idx + 1);//pick up everything after the first # as the binding key:
	}
	var binding = egl.eglx.services.$ServiceBinder.getBinding(eglddName.toLowerCase(), bindingKey);
	if(binding instanceof egl.eglx.services.RestBinding){
		ret = new egl.eglx.http.HttpRest();
		ret.request.uri = binding.baseURI;
		ret.restType = egl.eglx.rest.ServiceType.EglRpc;
	}
	else if(binding instanceof egl.eglx.services.DedicatedBinding){
		ret = new egl.eglx.http.HttpProxy();
	}
	if(ret === undefined || ret === null){
		throw egl.eglx.services.createServiceInvocationException("CRRUI3651E", [uri, eglddName]);  
	}
	return ret;
};
